polygon/sync/sync.go

```diff
@@ -211,33 +211,38 @@ func (s *Sync) applyNewBlockOnTip(
 		return nil
 	}
 
-	newHeaders := make([]*types.Header, len(newBlocks))
-	for i, block := range newBlocks {
-		newHeaders[i] = block.HeaderNoCopy()
+	headerChain := make([]*types.Header, len(blockChain))
+	for i, block := range blockChain {
+		block.Hash()
```

Comment:
Why is `block.Hash()` called here?

Reply:
good catch, I accidentally left it over after debugging